plugin.video.catchuptvandmore/resources/lib/channels/fr/6play.py

```diff
@@ -0,0 +1,382 @@
+
+
+url_json_video = 'https://pc.middleware.6play.fr/6play/v2/platforms/' \
+                 'm6group_web/services/6play/videos/%s'\
+                 '?csa=9&with=clips,freemiumpacks'
+
+
+url_img = 'https://images.6play.fr/v1/images/%s/raw'
+
+
+def channel_entry(params):
+    if 'list_shows' in params.next:
+        return list_shows(params)
+    elif 'list_videos' in params.next:
+        return list_videos(params)
+    elif 'play' in params.next:
+        return get_video_URL(params)
+
+
+@common.plugin.cached(common.cache_time)
+def list_shows(params):
+    shows = []
+
+    if params.next == 'list_shows_1':
+        file_path = utils.download_catalog(
+            url_root % (params.channel_name),
+            '%s.json' % (params.channel_name),
+            random_ua=True)
+        file_prgm = open(file_path).read()
+        json_parser = json.loads(file_prgm)
+
+        # do not cache failed catalog fetch
+        # the error format is:
+        #   {"error":{"code":403,"message":"Forbidden"}}
+        if isinstance(json_parser, dict) and \
+                'error' in json_parser.keys():
+            utils.os.remove(file_path)
+            raise Exception('Failed to fetch the 6play catalog')
+
+        for array in json_parser:
+            category_id = str(array['id'])
+            category_name = array['name'].encode('utf-8')
+            shows.append({
+                'label': category_name,
+                'url': common.plugin.get_url(
+                    action='channel_entry',
+                    category_id=category_id,
+                    next='list_shows_2',
+                    title=category_name
+                )
+            })
+
+        shows = common.plugin.create_listing(
+            shows,
+            sort_methods=(
+                common.sp.xbmcplugin.SORT_METHOD_UNSORTED,
+                common.sp.xbmcplugin.SORT_METHOD_LABEL
+            )
+        )
+
+    elif params.next == 'list_shows_2':
+        file_prgm = utils.get_webcontent(
+            url_category % (params.category_id),
+            random_ua=True)
+        json_parser = json.loads(file_prgm)
+
+        for array in json_parser:
+            program_title = array['title'].encode('utf-8')
+            program_id = str(array['id'])
+            program_desc = array['description'].encode('utf-8')
+            program_imgs = array['images']
+            program_img = ''
+            for img in program_imgs:
+                if img['role'].encode('utf-8') == 'vignette':
+                    external_key = img['external_key'].encode('utf-8')
+                    program_img = url_img % (external_key)
+                elif img['role'].encode('utf-8') == 'carousel':
+                    external_key = img['external_key'].encode('utf-8')
+                    program_fanart = url_img % (external_key)
+
+            info = {
+                'video': {
+                    'title': program_title,
+                    'plot': program_desc
```

Comment:
I don't understand, 'video' isn't the media type here ? I follow the Romanvm's simpleplugin documentation.
I paste the example below :

`listing = [{    'label': 'Label',
                'label2': 'Label 2',
                'thumb': 'thumb.png',
                'icon': 'icon.png',
                'fanart': 'fanart.jpg',
                'art': {'clearart': 'clearart.png'},
                'stream_info': {'video': {'codec': 'h264', 'duration': 1200},
                                'audio': {'codec': 'ac3', 'language': 'en'}},
                'info': {'video': {'genre': 'Comedy', 'year': 2005}},
                'context_menu': [('Menu Item', 'Action')],
                'url': 'plugin:/plugin.video.test/?action=play',
                'is_playable': True,
                'is_folder': False,
                'subtitles': ['/path/to/subtitles.en.srt', '/path/to/subtitles.uk.srt'],
                'mime': 'video/mp4'
                }]`